src/pipeline.cc

```diff
@@ -149,6 +155,84 @@ Result Pipeline::SetShaderCompileOptions(const Shader* shader,
                 shader->GetName());
 }
 
+Result Pipeline::SetShaderRequiredSubgroupSize(const Shader* shader,
+                                               const ShaderInfo::RequiredSubgroupSizeSetting subgroupSizeSetting) {
+  if (!shader)
+    return Result("invalid shader specified for  required subgroup size");
+
+  for (auto& info : shaders_) {
+    const auto* is = info.GetShader();
+    if (is == shader) {
+      info.SetRequiredSubgroupSizeSetting(subgroupSizeSetting);
+      return {};
+    }
+  }
+
+  return Result("unknown shader specified for required subgroup size: " +
+                shader->GetName());
+}
+
+Result Pipeline::SetShaderRequiredSubgroupSize(const Shader* shader,
+    const uint32_t subgroupSize)
+{
+    const bool isPow2 = (subgroupSize & (subgroupSize - 1)) == 0;
```

Comment:
If subgroupSize is 0 this is going underflow the `subgroupSize - 1`

Reply:
fixed